### PR TITLE
Timeout

### PR DESCRIPTION
**Pull request by thomedw**
This PR add configurable timeout

### CODE REVIEW

File: src/main/java/com/dubture/jenkins/digitalocean/Cloud.java [hunk 1]
```diff
@@ -329,7 +343,7 @@ public String getDisplayName() {
         public FormValidation doTestConnection(@QueryParameter String authToken) {
             try {
                 DigitalOceanClient client = new DigitalOceanClient(authToken);
-                client.getAvailableDroplets(1);
+                client.getAvailableDroplets(1, 10);
```
Comment (anpieber) [inline src/main/java/com/dubture/jenkins/digitalocean/Cloud.java]: this should be the timeout, shouldn't it?

File: src/main/java/com/dubture/jenkins/digitalocean/DigitalOcean.java [hunk 2]
```diff
@@ -112,7 +112,7 @@ public int compare(final Size s1, final Size s2) {
 
         do {
             page += 1;
-            images = client.getAvailableImages(page);
+            images = client.getAvailableImages(page, Integer.MAX_VALUE);
```
Comment (anpieber) [inline src/main/java/com/dubture/jenkins/digitalocean/DigitalOcean.java]: Without checking the API documentation - is this a timeout?

File: src/main/java/com/dubture/jenkins/digitalocean/DigitalOcean.java [hunk 3]
```diff
@@ -258,7 +258,7 @@ static String buildSizeLabel(final Size size) {
         int page = 1;
 
         do {
-            droplets = apiClient.getAvailableDroplets(page);
+            droplets = apiClient.getAvailableDroplets(page, Integer.MAX_VALUE);
```
Comment (anpieber) [inline src/main/java/com/dubture/jenkins/digitalocean/DigitalOcean.java]: Without checking the API documentation - is this a timeout?

**Comment by anpieber** [comment]
OK. I've checked the API myself - its ok

**Comment by nurupo** [comment]
So, looks like updating to the newer version of the `digitalocean-api-client` plugin broke some things in `src/main/java/com/dubture/jenkins/digitalocean/DigitalOcean.java`. For example, the regions for a droplet are being showed multiple times now, because the paging code behaves differently.

jenkins-do-plugin-bug

Similarly, SSH keys are displayed multiple times.

I'm testing the master branch only for a few minutes now, so far it looks like that only the paging code is broken.

---

Just for reference, where should plugin issues be submitted? I couldn't find  bug tracker on the [plugin page](https://wiki.jenkins-ci.org/display/JENKINS/DigitalOcean+Plugin) and the Issues seem to be disabled for this repo on GitHub, so I decided to comment on PR that broke things.

**Comment by anpieber** [comment]
Issues for all jenkins plugins are handled here: https://issues.jenkins-ci.org/secure/Dashboard.jspa - There's a component for the digital ocean plugin.

**Pull request merged**